cnd/src/network/comit_ln.rs

```diff
@@ -197,6 +198,26 @@ impl ComitLN {
         })
     }
 
+    // Once a swap has been finalized this removes it from the ComitLN cache of
+    // active swaps i.e., all the hash maps that we use to track state during
+    // the communication protocols.
+    fn remove_finalized_swap(&mut self, shared_id: SharedSwapId) {
```

Comment:
Wouldn't it be enough to remove the entry in `swaps_waiting_for_announcement`?

Reply:
Yes I think it most likely would, but for future proof-ness at the cost of some run time efficiency I figure why not clean the whole thing up.

Reply:
ok, is in the end I did as you suggest.  Because removing the swap from all the hash maps makes other code break. I could not get to the bottom of why, this makes me think that the code is currently brittle.  We have to re-think the whole hash map thing any way so I don't think its a big issue.